internal/cmd/create.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2022 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"chainguard.dev/vex/pkg/vex"
+)
+
+type createOptions struct {
+	vexDocOptions
+	vexStatementOptions
+	outFilePath string
+}
+
+// Validates the options in context with arguments
+func (o *createOptions) Validate(args []string) error {
+	if len(args) == 0 && len(o.Products) == 0 {
+		return errors.New("a required product id is required to generate a valid VEX statement")
+	}
+
+	if len(args) < 2 && o.Vulnerability == "" {
+		return errors.New("a vulnerability ID is required to generate a valid VEX statement")
+	}
+
+	if len(args) < 3 && o.Status == "" {
+		return fmt.Errorf("a valid impact status is required, one of %s", strings.Join(vex.Statuses(), ", "))
+	}
+
+	if len(args) >= 2 && o.Vulnerability != "" && args[1] != o.Vulnerability {
+		return errors.New("vulnerability can only be specified once")
+	}
+	if len(args) >= 3 && o.Status != "" && args[2] != o.Status {
+		return errors.New("status can only be specified once")
+	}
+
+	statusString := o.Status
+	if statusString == "" {
+		if len(args) < 3 {
+			return fmt.Errorf("a valid status is required to generate a valid VEX statement")
+		}
+		statusString = args[2]
+	}
+	status := vex.Status(statusString)
+	if !status.Valid() {
+		return fmt.Errorf(
+			"invalid VEX impact status '%s', valid status are: %s",
+			status, strings.Join(vex.Statuses(), ", "),
+		)
+	}
+
+	if status == vex.StatusNotAffected {
+		if o.Justification == "" {
+			return fmt.Errorf("an '%s' statement requires a valid justification: [%s]", vex.StatusAffected, strings.Join(vex.Justifications(), ", "))
+		}
+
+		if !vex.Justification(o.Justification).Valid() {
+			return fmt.Errorf("%s is not a valid VEX justification, valid justifications: %s", vex.StatusAffected, strings.Join(vex.Justifications(), ", "))
+		}
+	} else if o.Justification != "" {
+		return fmt.Errorf("a %s impact status must not have a justification", status)
+	}
+
+	return nil
+}
+
+func addCreate(parentCmd *cobra.Command) {
+	opts := createOptions{}
+	createCmd := &cobra.Command{
+		Short: fmt.Sprintf("%s create: creates a new VEX document", appname),
+		Long: fmt.Sprintf(`%s create: creates a new VEX document
+
+The create subcommand generates a single statement document
+from the command line. This is intended for simple use cases
+or to get a base document to get started.
+
+You can specify multiple products and customize the metadata of
+the document via the command line flags. vexctl will honor the
```

Comment:
It looks like you're using `%s` for the app name, did you want to use that for the "vexctl" on this line, too?